### ISSUE TITLE
OCIO won't compile with Visual Studio 2022

### ISSUE THREAD

**Issue by doug-walker** [AcademySoftwareFoundation/OpenColorIO#1634]
There may be a number of problems, but one has to do with PyBind 11:
https://github.com/pybind/pybind11/pull/3497

**Comment by remia** [comment]
Hi @doug-walker,

Do you have a build log to share? I'm a bit surprised as the analysis nightly build currently pass on [Visual Studio 2022](https://github.com/AcademySoftwareFoundation/OpenColorIO/runs/6121544585?check_suite_focus=true#step:11:20) as far as I know.

**Comment by doug-walker** [comment]
@remia , good point!  I did not realize that the Analysis is now using VS 2022, unlike CI which is still on VS 2019.  However, the Analysis is not building the Python binding:
https://github.com/AcademySoftwareFoundation/OpenColorIO/blob/b00877959f4497f9b8e5a08859d0ef8059fdd8e5/.github/workflows/analysis_workflow.yml#L361

So that's why it doesn't seem to be running into the PyBind 11 problem I mentioned.  

I notice the Analysis is installing PyBind 11 on Windows but then not using it (like it seems to for Linux and Mac).  Do you think Analysis should be building and testing the binding on Windows too?  @michdolan , any thoughts?

(We are working on trying to improve the Windows install instructions, so that's how we ran into the VS 2022 issue.)

**Comment by remia** [comment]
Yes that's a good catch, I'm not sure why it's setup like this but that was probably me during testing that noticed some issues with Python then forgot to go back and fixing it. I'll look into it when I get some time.

The next release, assuming it's a bump of VFX platform, will probably be a good opportunity to review all dependencies and increase minimal version where appropriate.

**Comment by remia** [comment]
I'm currently stuck trying to enable Python testing on the analysis build, there is a DLL load import [error](https://github.com/remia/OpenColorIO/runs/6290436198?check_suite_focus=true#step:13:3287) while running the Python unit test and I'm not able to understand where it comes from yet, it's very hard to debug using only Github Actions jobs. Note that it manages to successfully compile with pybind 2.6.2 on Visual Studio 22 so I assume it's a different issue than what you saw.

**Comment by doug-walker** [comment]
Yes, definitely a different issue.  The Pybind issue we encountered was only in Debug mode and the analysis builds in Release mode.
	https://github.com/pybind/pybind11/pull/3497
	https://github.com/microsoft/onnxruntime/issues/9735

**Comment by remia** [comment]
Should we bump the CI workflow to Visual Studio 2022 and update pybind11?

**Comment by doug-walker** [comment]
Remi, we have updated pybind in PR #1647.  

Regarding bumping CI to 2022, I think we should.  Especially since the draft CY2023 VFX Platform lists that as the minimum version: https://vfxplatform.com/

If no one objects, we will bump the CI to use Visual Studio 2022.
